indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -451,29 +451,51 @@ def _process_args(self):
+class RHRegistrationsExportPDF(RHRegistrationsExportBase):
+    """Export registration list to a PDF in table or book style."""
 
-    def _process(self):
-        pdf = RegistrantsListToPDF(self.event, reglist=self.registrations, display=self.export_config['regform_items'],
-                                   static_items=self.export_config['static_item_ids'])
-        try:
-            data = pdf.getPDFBin()
-        except Exception:
-            if config.DEBUG:
-                raise
-            raise NoReportError(_('Text too large to generate a PDF with table style. '
-                                  'Please try again generating with book style.'))
-        return send_file('RegistrantsList.pdf', BytesIO(data), 'application/pdf')
+    _base_spec = getattr(RHRegistrationsExportBase, 'normalize_url_spec', RH.normalize_url_spec)
 
+    normalize_url_spec = {
+        **_base_spec,
+        'preserved_args': (_base_spec.get('preserved_args', set()) | {'export_type'})
+    }
 
-class RHRegistrationsExportPDFBook(RHRegistrationsExportBase):
-    """Export registration list to a PDF in book style."""
+    def _process_args(self):
+        super()._process_args()
+        self.export_type = request.view_args.get('export_type')
 
     def _process(self):
-        static_item_ids, item_ids, _extra_item_ids = self.list_generator.get_item_ids()
-        pdf = RegistrantsListToBookPDF(self.event, self.regform, self.registrations, item_ids, static_item_ids)
-        return send_file('RegistrantsBook.pdf', BytesIO(pdf.getPDFBin()), 'application/pdf')
+        primary_headers, dynamic_headers, static_headers, rows = prepare_participant_list_data(
+            reglist=self.registrations,
+            display=self.export_config['regform_items'],
+            static_items=self.export_config['static_item_ids'],
+        )
+        generation_date = format_date(now_utc(), format='full', timezone=self.event.tzinfo)
+        if self.export_type not in ('table', 'book'):
+            raise BadRequest('Invalid export type')
+
+        if self.export_type == 'table':
+            css_path = 'events/registration/pdf/participants_table.css'
+            template_path = 'events/registration/pdf/participants_table.html'
+            filename = 'RegistrantsList'
+        else:  # book
+            css_path = 'events/registration/pdf/participants_book.css'
+            template_path = 'events/registration/pdf/participants_book.html'
+            filename = 'RegistrantsBook'
+
+        css = render_template(css_path)
+        html = render_template(
+            template_path,
+            event=self.event,
+            primary_headers=primary_headers,
+            dynamic_headers=dynamic_headers,
+            static_headers=static_headers,
+            rows=rows,
+            generation_date=generation_date
+        )
+        pdf = create_pdf(html, css, self.event)
+        return send_file(f'{filename}.pdf', pdf, 'application/pdf')
```

Comment:
please move the `.pdf` into the two `filename` vars above. I like being able to search for the filename and find the relevant code